pybossa/forms/forms.py

```diff
@@ -620,7 +620,7 @@ class UpdateProfileForm(Form):
     subscribed = BooleanField(lazy_gettext('Get email notifications'))
 
     locale = SelectField(lazy_gettext('Language'))
-    ckan_api = TextField(lazy_gettext('CKAN API Key'))
+    ckan_api = HiddenField(lazy_gettext('CKAN API Key'))
```

Comment:
Does changing this field affect the UI? If not, could the original code here be left as-is (and just hide the HTML from the template)? This would avoid needing to make a change to the pybossa server-side code.

Reply:
This field is from the WTForms library which renders when the form is instantiated so it isn't in the HTML: 

"Field instances contain the data of that instance as well as the functionality to render it within your Form. They also contain a number of properties which can be used within your templates to render the field and label."
https://wtforms.readthedocs.io/en/2.3.x/fields/

Using the HiddenField ensures that the validations that require the field still pass while not showing it in the UI.

Reply:
This makes sense. I would still confirm if just removing the HTML from the template is sufficient, or if this change is needed as well. The idea is that if we ever decide to bring this field back, we would only need to restore the HTML and not require changing the server-side code. However, if both changes are necessary, then it makes sense. Thanks!

Reply:
I agree that it make sense to just hide ckan-api from UpdateProfileForm, since HTML is rendering the `UpdateProfileForm` as a whole, making it hard to remove a single field.